File: 200/283 - Move Zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int z = 0, nz = 0;
    
    while (z < numsSize && nz < numsSize) {
        if (nums[z] != 0) {
            nz = ++z;
            continue;
        }
        
        if (nums[nz] == 0) {
            ++nz;
            continue;
        }
        
        nums[z++] = nums[nz];
        nums[nz++] = 0;
    }
}

File: csharp/283.move-zeroes.cs
/*
 * @lc app=leetcode id=283 lang=csharp
 *
 * [283] Move Zeroes
 */
public class Solution {
    public void MoveZeroes(int[] nums) {
        int n = nums.Length, zero = 0;
        for (int i = 0; i < n; ++i) {
            int num = nums[i];
            if (num != 0) {
                if (zero != i) {
                    nums[zero] = num;
                    nums[i] = 0;
                }
                ++zero;
            }
        }
    }
}

